structure_tests/types.go

```diff
@@ -1,86 +1,89 @@
 package main
 
 import (
-	"encoding/json"
-	"io/ioutil"
 	"strings"
-
-	"github.com/ghodss/yaml"
+	"testing"
 )
 
-type arrayFlags []string
-
-func (a *arrayFlags) String() string {
-	return strings.Join(*a, ", ")
+type StructureTest interface {
+	RunAll(t *testing.T)
 }
 
-func (a *arrayFlags) Set(value string) error {
-	*a = append(*a, value)
-	return nil
+type CommandTest interface {
```

Comment:
eh not really, just had to move the `validate` methods to pair with the struct definitions.